src/compiler/worker/getConfig.js

```diff
@@ -14,10 +14,15 @@ module.exports = function getConfig(
   configOptions: any,
   platform: Platform
 ) {
-  return makeReactNativeConfig(
+  let config;
+  try {
     // $FlowFixMe
-    require(configPath),
-    configOptions,
-    platform
-  );
+    config = require(configPath);
+  } catch (e) {
+    throw new Error(
+      'Haul configuration cannot be loaded. Have you provided valid JS file?'
+    );
+  }
+
+  return makeReactNativeConfig(config, configOptions, platform);
```

Comment:
This should do something like `config.__esModule ? config.default : config` so we support both ES exports and node exports.

Reply:
https://github.com/callstack/haul/blob/fde95a923d288b91c3fdeba301275a8692123b7d/src/utils/makeReactNativeConfig.js#L236-L237

Reply:
We should use the `__esModule` check for this  since that's what Babel does.

Reply:
Yeah, check for `__esModule` makes sense. It's definitely cleaner. 👍